thirdeye-ui/src/app/components/subscription-group-wizard/alerts-dimensions/add-dimension-dialog/add-dimension-dialog.component.tsx

```diff
@@ -0,0 +1,183 @@
+            ? getAssociationIdsForAlert({
+                  enumerationIds: enumerationItemsForAlert.map(
+                      (item) => item.id
+                  ),
+                  alertId: selectedAlertId,
+              })
+            : [];
+
+        const handleChangeAssociations = (
+            newAssociationIds: string[]
+        ): void => {
+            associationIdsForAlert.forEach((associationId) => {
+                setAssociationsSelected((stateProp) => ({
+                    ...stateProp,
+                    [associationId]: newAssociationIds.includes(associationId),
+                }));
+            });
+        };
+
+        return (
+            <Box>
+                <Typography variant="h6">{t("label.alert")}</Typography>
+
+                <InputSection
+                    fullWidth
+                    inputComponent={
+                        <Autocomplete<Alert>
+                            getOptionLabel={(option) => option.name}
+                            options={alerts}
+                            renderInput={(params) => (
+                                <TextField
+                                    {...params}
+                                    InputProps={{
+                                        ...params.InputProps,
+                                    }}
+                                    placeholder={t("message.select-entity", {
+                                        entity: t("label.alert"),
+                                    })}
+                                    variant="outlined"
+                                />
+                            )}
+                            value={selectedAlert}
+                            onChange={(_, alertProp) => {
+                                if (!alertProp) {
+                                    return;
+                                }
+
+                                setSelectedAlertId(alertProp.id);
+                            }}
+                        />
+                    }
+                    label={t("message.select-entity", {
+                        entity: t("label.alert"),
+                    })}
+                />
+
+                <EmptyStateSwitch
+                    emptyState={null}
+                    isEmpty={!(selectedAlert && enumerationItems)}
+                >
+                    {selectedAlert && enumerationItems ? (
+                        <>
+                            <Box pb={2} pt={2}>
+                                <Divider />
+                            </Box>
+
+                            <AlertsTable
+                                associations={associationsForAlert}
+                                enumerationItemsForAlert={
+                                    enumerationItemsForAlert
+                                }
+                                selectedAlert={selectedAlert}
+                                onChangeAssociations={handleChangeAssociations}
```

Comment:
we should cast `selectedAlert as Alert` and `enumerationItems as EnumerationItem[]` so we don't have to do the check again

Reply:
That seems simpler. Updated.